plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreConfig.java

```diff
@@ -339,4 +348,33 @@ public ThriftMetastoreConfig setBatchMetadataFetchEnabled(boolean batchMetadataF
         this.batchMetadataFetchEnabled = batchMetadataFetchEnabled;
         return this;
     }
+
+    public String getHttpBearerToken()
+    {
+        return httpBearerToken;
+    }
+
+    @Config("hive.metastore.http.client.bearer-token")
+    @ConfigSecuritySensitive
+    @ConfigDescription("Http Bearer token to authenticate with a HTTP transport based metastore service")
+    public ThriftMetastoreConfig setHttpBearerToken(String httpBearerToken)
+    {
+        this.httpBearerToken = httpBearerToken;
+        return this;
+    }
+
+    public Map<String, String> getAdditionalHeaders()
+    {
+        return additionalHeaders;
+    }
+
+    @Config("hive.metastore.http.client.additional-headers")
+    @ConfigDescription("Comma separated key:value pairs to be send to metastore as additional headers")
```

Comment:
What happens if we want to include `,` or `:` in the header key or value?

Reply:
https://github.com/trinodb/trino/blob/675defb40e12734f5d067848a14857ff66c80019/plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListenerConfig.java#L102-L116 

Let's use this existing functionality.

For the concerns from @ebyhr while using the above mentioned functionality:  https://trino.io/docs/current/admin/event-listeners-http.html?highlight=custom+http+headers#custom-http-headers

Reply:
Replaced my implementation with the logic similar to above.